pkg/fpga/ioctl_linux.go

```diff
@@ -19,9 +19,14 @@ import (
 	"syscall"
 )
 
-func ioctl(fd uintptr, req uint, arg uintptr) (ret uintptr, err error) {
-	ret, _, err = syscall.Syscall(syscall.SYS_IOCTL, fd, uintptr(req), arg)
-	return
+func ioctl(fd uintptr, req uint, arg uintptr) (uintptr, error) {
+	ret, _, err := syscall.Syscall(syscall.SYS_IOCTL, fd, uintptr(req), arg)
+	// Even though err is syscall.Errno which implements the error interface it's
+	// an unsigned number and can't be nil. So filter the zero value out.
+	if err == 0 {
```

Comment:
```
ret, _, err := syscall.Syscall(syscall.SYS_IOCTL, fd, uintptr(req), arg)
if err != 0 {
   return ret, err
}
return ret, nil
```
so, normal return in the end of function, error return condition inside `if`

See best practices example from golang libraries here: https://github.com/golang/sys/blob/master/unix/syscall_linux.go#L77

Reply:
In this particular case I'd disagree. Filtering out 0 errno with the comment is more readable this way in my opinion.

Reply:
@bart0sh comment can be added, but normal return inside `if` is anti-pattern.
normal exit from the function should not be hidden inside `if`.

Reply:
I wouldn't call this an example of best practices. My editor indicates that `Syscall()` already returns an object satisfying the `error` interface. The first natural reaction is to assume the principle of least astonishment is adhered by the standard library and to return it as it is. But hey... you can't do that without consulting with the doc.

> normal return inside if is anti-pattern

This is interesting. Could you please refer it?

Reply:
I have seen several examples where a normal return is inside an if, for example to return early from a function that doesn't need to do anything. I don't consider it an anti-pattern.

Reply:
@rojkov I prefer not to re-invent the wheel, and use same patterns that are part of standard library. Once x/sys/unix will have needed implementation of `ioctl`, this code can be dropped. meanwhile, it is good to have it as close as possible to code in x/sys/unix, similar to one I referenced above.

Reply:
Updated